src/geom/Geometry.cpp

```diff
@@ -595,7 +597,13 @@ Geometry::Union() const
 #ifdef DISABLE_OVERLAYNG
     return UnaryUnionOp::Union(*this);
 #else
-    return operation::overlayng::OverlayNGRobust::Union(this);
+
+    operation::cluster::GeometryIntersectsClusterFinder f;
+    operation::cluster::DisjointOperation op(f);
+
+    return op.processDisjointSubsets(*this, [](const Geometry& g) {
```

Comment:
If this is superior, it would be better to provide it as a separate/new operation class, rather than burying it under Geometry implementation.

Reply:
Also, this is new implementation for a widely-used operation.  So better to put it in a separate PR, in case of issues.

Reply:
Would be good to see metrics on when and how this improves performance.  Is there any negative affect for cases which have no disjoint geometry?

Reply:
is this safe when Overlay invokes snapping or snap-rounding?  That can cause overlay results to intersect even when the input is disjoint.

Reply:
This passes all existing test cases but I suspect that it is not safe. Can you point me to an example of disjoint inputs producing an intersecting result so I can try to provoke a failure?

Reply:
Unfortunately I don't have a test case for this situation.  I think I have seen it in the wild, however.

Reply:
Could the overlay short-circuits for disjoint inputs be similarly affected? Like `OverlayUtil::isEmptyResult` ?

Reply:
No.  When snap-rounding is required explicitly by providing a precision model, the disjoint checks are done using that precision model.  And for full precision invocations, the disjoint check is accurate as is.  

The risk occurs when a robustness error forces snapping or snap-rounding to be used internally.  In this case the envelope of the result can be larger than the envelope of the inputs.

Reply:
I guess the workaround would be to cluster based on distance. This currently has some severe performance problems that I am working to address (#691)

Reply:
Or, to verify that the results of the sub-unions remain disjoint.